### PR TITLE
CR#22502 Fix duplicate flag for convert tool

### PR DESCRIPTION
**Pull request by DreierF**
Addresses issue #

- [ ] Changes are tested adequately
- [ ] Agent's README.md updated in case of user-visible changes
- [ ] CHANGELOG.md updated

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

File: agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java [hunk 1]
```diff
@@ -125,14 +127,14 @@ public int getSplitAfter() {
...
+	/** @see #duplicateClassFileBehavior */
+	public EDuplicateClassFileBehavior getDuplicateClassFileBehavior() {
+		return duplicateClassFileBehavior;
 	}
 
-	/** @see #shouldIgnoreDuplicateClassFiles */
-	public void setShouldIgnoreDuplicateClassFiles(boolean shouldIgnoreDuplicateClassFiles) {
-		this.shouldIgnoreDuplicateClassFiles = shouldIgnoreDuplicateClassFiles;
+	/** @see #duplicateClassFileBehavior */
+	public void setDuplicateClassFileBehavior(EDuplicateClassFileBehavior duplicateClassFileBehavior) {
```
Comment (karottenreibe) [inline agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java:136]: seems to be unused. delete?

**Comment by karottenreibe** [comment]
@DreierF one more thing: did you check the README? do we need to update it?

**Comment by DreierF** [comment]
README would not have needed an update, but I took this as an opportunity to also replace the ignore-duplicates in the agent with `duplicates` and `filter`-> `includes` to match the argument names of the agent

File: agent/README.md [hunk 2]
```diff
@@ -81,9 +81,9 @@ patterns with `*`, `**` and `?`.
 - `interval`: the interval in minutes between dumps of the current coverage to an XML file (Default is 60). If set to 
   0 coverage is only dumped at JVM shutdown.
 - `dump-on-exit`: whether a coverage report should be written on JVM shutdown (Default is true).
-- `ignore-duplicates`: forces JaCoCo to ignore duplicate class files. This is the default to make the initial
-  setup of the tool as easy as possible. However, this should be disabled for productive use if possible.
-  See the special section on `ignore-duplicates` below.
+- `duplicates`: defines how JaCoCo handles duplicate class files. This is by default set to WARN to make the initial
```
Comment (karottenreibe) [inline agent/README.md:84]:
```suggestion
- `duplicates`: defines how JaCoCo handles duplicate class files. This is by default set to `WARN` to make the initial
```

File: agent/README.md [hunk 3]
```diff
@@ -81,9 +81,9 @@ patterns with `*`, `**` and `?`.
...
+  setup of the tool as easy as possible. However, this should be set to FAIL for productive use if possible. In special
```
Comment (karottenreibe) [inline agent/README.md:85]:
```suggestion
  setup of the tool as easy as possible. However, this should be set to `FAIL` for productive use if possible. In special 
```

File: agent/README.md [hunk 4]
```diff
@@ -81,9 +81,9 @@ patterns with `*`, `**` and `?`.
 - `interval`: the interval in minutes between dumps of the current coverage to an XML file (Default is 60). If set to 
   0 coverage is only dumped at JVM shutdown.
 - `dump-on-exit`: whether a coverage report should be written on JVM shutdown (Default is true).
-- `ignore-duplicates`: forces JaCoCo to ignore duplicate class files. This is the default to make the initial
-  setup of the tool as easy as possible. However, this should be disabled for productive use if possible.
-  See the special section on `ignore-duplicates` below.
+- `duplicates`: defines how JaCoCo handles duplicate class files. This is by default set to WARN to make the initial
+  setup of the tool as easy as possible. However, this should be set to FAIL for productive use if possible. In special 
+  cases you can also set it to IGNORE to print no warnings in such cases. See the special section on `duplicates` below.
```
Comment (karottenreibe) [inline agent/README.md:86]:
```suggestion
  cases you can also set it to `IGNORE` to print no warnings. See the special section on `duplicates` below.
```

**Pull request merged**